tests/interfaces-home/task.yaml

```diff
@@ -0,0 +1,147 @@
+
+  echo "When the plug is connected"
+  sudo snap connect $SNAP_NAME:$PLUG ubuntu-core:$PLUG
+  echo "$(snap interfaces)" | grep -Pzq "$CONNECTED_PATTERN"
+
+  echo "Then the snap is able to read home files"
+  response=$(home-consumer.reader $READABLE_FILE)
+  echo "$response" | grep -Pqz "$OK_CONTENT"
+
+  echo "============================================"
+
+  echo "When the plug is disconnected"
+  sudo snap disconnect $SNAP_NAME:$PLUG ubuntu-core:$PLUG
+  echo "$(snap interfaces)" | grep -Pzq "$DISCONNECTED_PATTERN"
+
+  echo "Then snap can't read home files"
+  if home-consumer.reader $READABLE_FILE; then
+    echo "Home files shouldn't be readable" && exit 1
+  fi
+
+  echo "============================================"
+
+  echo "When the plug is connected"
+  sudo snap connect $SNAP_NAME:$PLUG ubuntu-core:$PLUG
+  echo "$(snap interfaces)" | grep -Pzq "$CONNECTED_PATTERN"
+
+  echo "Then the snap is able to append to home files"
+  home-consumer.writer "$WRITABLE_FILE"
+  cat "$WRITABLE_FILE" | grep -Pqz "${PREVIOUS_CONTENT}\n${OK_CONTENT}"
+
+  echo "============================================"
+
+  echo "When the plug is disconnected"
+  sudo snap disconnect $SNAP_NAME:$PLUG ubuntu-core:$PLUG
+  echo "$(snap interfaces)" | grep -Pzq "$DISCONNECTED_PATTERN"
+
+  echo "Then snap can't append to home files"
+  if home-consumer.writer "$WRITABLE_FILE"; then
+    echo "Home files shouldn't be writable" && exit 1
+  fi
+
+  echo "============================================"
+
+  echo "When the plug is connected"
+  sudo snap connect $SNAP_NAME:$PLUG ubuntu-core:$PLUG
+  echo "$(snap interfaces)" | grep -Pzq "$CONNECTED_PATTERN"
+
+  echo "Then the snap is able to create home files"
+  home-consumer.writer "$CREATABLE_FILE"
+  cat "$CREATABLE_FILE" | grep -Pqz "$OK_CONTENT"
+
+  echo "============================================"
+
+  echo "When the plug is disconnected"
+  sudo snap disconnect $SNAP_NAME:$PLUG ubuntu-core:$PLUG
+  echo "$(snap interfaces)" | grep -Pzq "$DISCONNECTED_PATTERN"
+
+  echo "Then snap can't create home files"
+  if home-consumer.writer "$CREATABLE_FILE"; then
+    echo "Home shouldn't be accessible" && exit 1
+  fi
+
+  echo "============================================"
+
+  echo "When the plug is connected"
+  sudo snap connect $SNAP_NAME:$PLUG ubuntu-core:$PLUG
+  echo "$(snap interfaces)" | grep -Pzq "$CONNECTED_PATTERN"
+
+  echo "Then the snap is not able to read hidden home files"
+  if home-consumer.reader "$HIDDEN_READABLE_FILE"; then
+    echo "Home shouldn't be accessible" && exit 1
+  fi
+
+  echo "============================================"
+
+  echo "When the plug is connected"
+  sudo snap connect $SNAP_NAME:$PLUG ubuntu-core:$PLUG
+  echo "$(snap interfaces)" | grep -Pzq "$CONNECTED_PATTERN"
+
+  echo "Then the snap is not able to write hidden home files"
+  if home-consumer.writer "$HIDDEN_CREATABLE_FILE"; then
+    echo "Home shouldn't be accessible" && exit 1
```

Comment:
Hmm, `It should be impossible to create hidden files`  something something, just the copy-pasted error message doesn't quite fit each case.